### PR TITLE
Change abundances to mass fraction

### PR DESCRIPTION
**Pull request by jvshields**
### :pencil: Description

**Type:** | :rocket: `feature` | :roller_coaster: `infrastructure`

This PR should probably have been broken into two pull requests. It does two primary things.

marcs.py has been changed to add a to_composition() method that uses the Composition class added in [PR 103 ](https://github.com/tardis-sn/stardis/pull/103). It also adds a to_stellar_model() method that outputs a geometry, composition, and temperatures, though this method is currently not called anywhere. 

The second thing this PR does is alter the StellarModel class to hold the composition, allowing for the previous to_stellar_model() method to exist, and updates read_marcs_to_fv() correctly to handle this change. 

The read_marcs_model import in stardis/model/base.py was moved inside read_marcs_to_fv() to avoid a circular import. This is necessary for now, but should not persist because we aim to remove the read_marcs_to_fv() function altogether and just move over to the new model reader that can directly output a stellar model that can be correctly handled by stardis. 

### :vertical_traffic_light: Testing

How did you test these changes?

- [ ] Testing pipeline
- [ ] Tested the quickstart notebook locally to make sure the code still runs and produces the expected output. 

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

File: stardis/model/base.py [hunk 1]
```diff
@@ -56,10 +54,14 @@ def read_marcs_to_fv(fpath, atom_data, final_atomic_number=30):
     stardis.model.base.StellarModel
     """
 
+    # This is a hacky workaround to avoid circular imports. Should be fixed when we remove the read_marcs_to_fv function.
+    from stardis.io.model.marcs import read_marcs_model
```
Comment (sonachitchyan) [inline stardis/model/base.py:58]: why is this not in the beginning?

**Review by sonachitchyan — APPROVED** [review]

File: stardis/io/model/marcs.py [hunk 2]
```diff
@@ -31,20 +35,95 @@ def to_geometry(self):
...
-    #     atomic_mass_fraction = self.data[
-    #         [f"scaled_log_number_fraction_{i+1}" for i in range(30)]
-    #     ].values
-    #     return Composition(density, atomic_mass_fraction)
+    def to_composition(self, atom_data, final_atomic_number=30):
+        """
+        Returns a stardis.model.composition.base.Composition object from the MARCS model.
+
+        Returns
+        -------
```
Comment (atharva-2001) [inline stardis/io/model/marcs.py:43]: Parameter docstring

Comment (atharva-2001) [inline stardis/io/model/marcs.py:43]: Parameter docstring

File: stardis/io/model/marcs.py [hunk 3]
```diff
@@ -31,20 +35,95 @@ def to_geometry(self):
...
+                > atom_data.atom_data.index.max()
+            ):
+                logging.warning(
+                    f"Final model chemical number is {len(marcs_chemical_mass_fractions.columns)} while final atom data chemical number is {atom_data.atom_data.index.max()} and final atomic number requested is {final_atomic_number}."
+                )
+
+        for atom_num, col in enumerate(marcs_chemical_mass_fractions.columns):
+            if atom_num >= len(atom_data.atom_data):
+                marcs_chemical_mass_fractions[f"mass_fraction_{atom_num+1}"] = np.nan
+            else:
```
Comment (DeerWhale) [inline stardis/io/model/marcs.py:71]: Break the loop for the rest of the elements that larger than len(atom_data)

File: stardis/io/model/marcs.py [hunk 4]
```diff
@@ -31,20 +35,95 @@ def to_geometry(self):
...
-    #     -------
-    #     stardis.model.composition.base.Composition
-    #     """
-    #     density = self.data.density.values * u.g / u.cm**3
-    #     # THIS IS VERY WRONG. NEEDS TO BE FIXED WITH WORK FROM THE MASS FRACTION BRANCH
-    #     atomic_mass_fraction = self.data[
-    #         [f"scaled_log_number_fraction_{i+1}" for i in range(30)]
-    #     ].values
-    #     return Composition(density, atomic_mass_fraction)
+    def to_composition(self, atom_data, final_atomic_number=30):
```
Comment (DeerWhale) [inline stardis/io/model/marcs.py:38]: Maybe get rid of the default for the final_atomic_number.

File: stardis/io/model/marcs.py (hunk 2 above)
Comment (jvshields) [reply stardis/io/model/marcs.py:43]: Fixed

File: stardis/io/model/marcs.py [hunk 5]
```diff
@@ -31,20 +35,95 @@ def to_geometry(self):
...
+
+        for atom_num, col in enumerate(marcs_chemical_mass_fractions.columns):
+            if atom_num >= len(atom_data.atom_data):
+                marcs_chemical_mass_fractions[f"mass_fraction_{atom_num+1}"] = np.nan
+            else:
+                marcs_chemical_mass_fractions[f"mass_fraction_{atom_num+1}"] = (
+                    10 ** marcs_chemical_mass_fractions[col]
+                ) * atom_data.atom_data.mass.iloc[atom_num]
+
+        marcs_chemical_mass_fractions[
```
Comment (DeerWhale) [inline stardis/io/model/marcs.py:76]: This part seems redundant.

File: stardis/io/model/marcs.py [hunk 6]
```diff
@@ -31,20 +35,95 @@ def to_geometry(self):
...
+            f"mass_fraction_{atom_num+1}"
+        ] = marcs_chemical_mass_fractions[
+            f"mass_fraction_{atom_num+1}"
+        ] / marcs_chemical_mass_fractions.filter(
+            regex="mass_fraction"
+        ).sum(
+            axis=1
+        )
```
Comment (DeerWhale) [inline stardis/io/model/marcs.py:84]: This part seems redundant.

**Review by DeerWhale — APPROVED** [review]

**Pull request merged**